aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -248,8 +256,9 @@ c10::optional<int64_t> NestedTensorImpl::opt_size(int64_t d) const {
   return (*opt_sizes_)[d];
 }
 
-void NestedTensorImpl::refresh_dim() {
-  const auto my_dim = nested_sizes_.dim() ? nested_sizes_.sizes()[1] + 1 : 1;
+void NestedTensorImpl::refresh_dim(const c10::optional<int64_t> dim_) {
+  const auto my_dim = dim_.has_value() ? *dim_ :
+      (nested_sizes_.dim() ? nested_sizes_.size(1) + 1 : 1);
   sizes_and_strides_.resize(my_dim);
   TORCH_INTERNAL_ASSERT_DEBUG_ONLY(dim() == my_dim);
```

Comment:
This is supposed to be static right? So you can unconditionally call `sym_size(1)` on `nested_sizes_` and guard on it. Then you don't need the API change.